### PR TITLE
Scikit Learn update to 1.2.1

### PR DESCRIPTION
**Pull request by dimitrismistriotis**

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     strategy:
       matrix:
         # For quotes on 3.10: https://github.com/actions/setup-python/issues/160
-        python_version: [3.8, 3.9, '3.10']
+        python_version: [3.8, 3.9, '3.10', 3.11]
```
Comment (menkotoglou) [inline .github/workflows/checks.yml:12]:
```suggestion
        python_version: [3.8, 3.9, '3.10', '3.11']
```
If I remember the root cause correctly, we should probably add these to 3.11 too.

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
...
```
Comment (menkotoglou) [inline .github/workflows/checks.yml:27]: Same here

**Review by menkotoglou — APPROVED** [review]
Soft approve. LGTM!

File: .github/workflows/checks.yml (hunk 1 above)
Comment (dimitrismistriotis) [reply .github/workflows/checks.yml:12]: Reason for not putting them (in Python, think it is universal):

```python
>>> float("3.10")
3.1
>>> float("3.11")
3.11
```

if not `.0` then quotes are not needed. Left it for Python 3.20 :smile:

**Review by menkotoglou — APPROVED** [review]
Reviewed comments and you're right :heavy_check_mark: . Let's go for it!

**Pull request merged**